### PR TITLE
Remove default runAsUser

### PR DESCRIPTION
**Pull request by Fodoj**
**Why This PR?**

Same as in https://github.com/FairwindsOps/charts/pull/271/files - fixed runAsUser breaks deployments on OpenShift. Alternative, I can adjust the code to allow skipping security context completely.

### CODE REVIEW

**Comment by sudermanjr** [comment]
You'll want to bump the chart version, rather than the `appVersion`.  App version dictates the image to pull. 

Additionally, I think it would be great if we could move to a completely configurable security context like you mentioned in the original description. That's the pattern I've started using on newer charts. That way we can leave the defaults as what they are now, but make the chart a lot more flexible going forward.

**Comment by Fodoj** [comment]
@sudermanjr yep, I actually changed to allow completely skip SC for containers right after I created PR :) Updated, both implementation and adjusted the version.

**Comment by Fodoj** [comment]
I can not access failed builds to see what exactly went wrong there, some hints appreciated

**Comment by sudermanjr** [comment]
Don't worry about the insights ones. Can you not see the ones in Circle though?

**Comment by sudermanjr** [comment]
```
[ERROR] templates/: parse error at (goldilocks/templates/dashboard-deployment.yaml:74): unexpected EOF
```

**Comment by sudermanjr** [comment]
You can run `helm lint` on the chart to catch those before you push if that makes it easier.

**Comment by Fodoj** [comment]
Login page of CircleCI confused me. Pushed the fix. 

I plan to add support for extraContainers for dashboard (so that it's possible to put oauth-proxy in front, for example) - is it ok to add externals dependencies like https://github.com/bitnami/charts/tree/master/bitnami/common to Fairwind Charts?

**Comment by sudermanjr** [comment]
I would prefer to avoid adding any external sub-charts here since we can't control the changes that they make.

However, if it could be done in a way that is disabled by default, that might be okay

File: stable/goldilocks/templates/controller-deployment.yaml [hunk 1]
```diff
@@ -41,8 +41,10 @@ spec:
             {{- range $name, $value := .Values.controller.flags }}
             - --{{ $name }}={{ $value }}
             {{- end }}
+        {{- if .Values.controller.securityContext }}
```
Comment (sudermanjr) [inline stable/goldilocks/templates/controller-deployment.yaml:44]: Just realizing now that we already had the securityContext in here. Does setting `controller.securityContext: {}` not effectively do this in the existing version of the chart?

Comment (Fodoj) [reply stable/goldilocks/templates/controller-deployment.yaml:44]: It doesn't - Helm will deep merge two hashes, so everything from the default values.yaml will still be there

Comment (sudermanjr) [reply stable/goldilocks/templates/controller-deployment.yaml:44]: Interesting. I hadn't thought about that. 👍🏼

**Review by sudermanjr — APPROVED** [review]

**Pull request merged**